src/Facets/CalldataVerificationFacet.sol

```diff
@@ -117,10 +122,23 @@ contract CalldataVerificationFacet {
         )
     {
         LibSwap.SwapData[] memory swapData;
-        (, , , receiver, receivingAmount, swapData) = abi.decode(
-            data[4:],
-            (bytes32, string, string, address, uint256, LibSwap.SwapData[])
-        );
+        if (
+            abi.decode(data, (bytes4)) ==
+            StandardizedCallFacet.standardizedCall.selector
+        ) {
+            // standardizedCall
+            bytes memory unwrappedData = abi.decode(data[4:], (bytes));
+            (, , , receiver, receivingAmount, swapData) = abi.decode(
+                unwrappedData.slice(4, unwrappedData.length - 4),
+                (bytes32, string, string, address, uint256, LibSwap.SwapData[])
+            );
+        } else {
+            // normal call
+            (, , , receiver, receivingAmount, swapData) = abi.decode(
+                data[4:],
+                (bytes32, string, string, address, uint256, LibSwap.SwapData[])
```

Comment:
This is duplicated code and could be optimized. 
the abi.decode() is exactly the same, just the bytes parameter differs. So the if statement could just prepare the bytes parameter and then the decode follows after the if statement